### PR TITLE
Combined work

### PR DESCRIPTION
**Pull request by RaghvindYadav**
In this PR we tried to solve the following things:

1. Get in Rupali's updates for the account settings, Add the "generate " username button to the username input (@rupalirajput2597 )
2. Update the reset password and sign in form to use the same error handling at sign up (@RaghvindYadav )
3. QA and address edge cases, make sure we show all relevant errors from the client, etc. (@RaghvindYadav )
4. Remove any other remaining components we no longer need (@rupalirajput2597 )
5. Write unit & integration tests (@RaghvindYadav )

For the 5the points I have some doubts so I write in the slack for it.

### CODE REVIEW

File: src/lib/contexts/msa-listener-handler.svelte.ts [hunk 1]
```diff
@@ -88,6 +89,7 @@ export class MsaListenerHandler {
               action.notificationResult,
             );
             errorMessage = 'We could not verify the token you entered. Please try again.';
+            if (this.onFailure) this.onFailure();
```
Comment (jonahtwalker) [inline src/lib/contexts/msa-listener-handler.svelte.ts:92]: What callback would you want to run when the token is incorrrect?

File: src/routes/reset-password/+page.svelte [hunk 2]
```diff
@@ -5,7 +5,7 @@
   let { data }: { data: PageData } = $props();
 </script>
 
-<div class="flex h-full w-full items-center justify-center px-4">
+<div class="flex h-full w-full items-center justify-center p-4">
```
Comment (jonahtwalker) [inline src/routes/reset-password/+page.svelte]: Why? Why make this page layout any different than the other two identical layouts?

File: src/routes/reset-password/reset-password-form.svelte [hunk 3]
```diff
@@ -59,17 +61,18 @@
...
-
       debounceTimer = window.setTimeout(async () => {
         try {
+          loading = true;
           const result = await validateForm({ update: true });
 
           // Check if identifier is available for step 1
           if (step === 1 && $formData.ident) {
             const isIdentValid = await checkIdentAvailability();
+            console.log('Result valid', result.valid, 'isIdentValid', isIdentValid);
```
Comment (jonahtwalker) [inline src/routes/reset-password/reset-password-form.svelte]: Let's be sure to clean up logs like these

File: src/routes/signin/+page.svelte [hunk 4]
```diff
@@ -5,7 +5,7 @@
   let { data }: { data: PageData } = $props();
 </script>
 
-<div class="flex h-full w-full items-center justify-center px-4">
+<div class="flex h-full w-full items-center justify-center p-4">
```
Comment (jonahtwalker) [inline src/routes/signin/+page.svelte]: Again, why? What about the Sign Up page - then?

File: src/routes/reset-password/reset-password-form.svelte [hunk 5]
```diff
@@ -379,12 +362,12 @@
         return true; // Account exists, which is what we want
       } else {
         // If identifier IS available, it means no account exists with this identifier
-        updateErrorMessage('No account found with this identifier.', 'ident');
+        updateFormErrors('ident', 'No account found with this identifier.');
```
Comment (jonahtwalker) [inline src/routes/reset-password/reset-password-form.svelte]: We talked about these stories, and we do not want to inform the user. We are feigning success and leaving it to them to deduce if the email doesn't arrive.

We will need to add forgot username later.

File: src/routes/signup/sign-up-form.svelte [hunk 6]
```diff
@@ -344,13 +344,17 @@
     try {
       isLoading = true;
       const result = await myUserContext.findAvailableUserHandle($formData.email);
+      if (result && typeof result === 'object' && 'object' in result) {
+        $formData.username = result.object ?? '';
```
Comment (jonahtwalker) [inline src/routes/signup/sign-up-form.svelte]: Why is this necessary?

File: src/routes/signup/sign-up-form.svelte [hunk 7]
```diff
@@ -437,6 +441,9 @@
           fieldName="username"
           placeholder="e.g. 'giraffe08'"
           label="Username"
+          {identType}
+          generateUsername={getSuggestedUsername}
```
Comment (jonahtwalker) [inline src/routes/signup/sign-up-form.svelte:442]: Why do we need to pass this to identInput? We set it right in the formData above in `getSuggestedUsername`

**Review by jonahtwalker — CHANGES_REQUESTED** [review]

**Comment by jonahtwalker** [comment]
I pushed a commit that cleans up a bunch of my comments I left above. I am not sure yet what else has been changed in the Sign Up Form that causes the validation error to come from Zod instead of our schema, but there's still more to do. The button state is weird as you changed the way the methods are loading, too.

There's so much QA to do, it doesn't seem like you did much before you submitted. I didn't look at your stories at all since the components still aren't yet working as expected.

I am merging this in with these issues, we have too many branches.

File: src/routes/reset-password/+page.svelte (hunk 2 above)
Comment (RaghvindYadav) [reply src/routes/reset-password/+page.svelte]: I made this changes for mobile view, top border of AuthCard was sticked to the header.

File: src/lib/contexts/msa-listener-handler.svelte.ts (hunk 1 above)
Comment (RaghvindYadav) [reply src/lib/contexts/msa-listener-handler.svelte.ts:92]: I tested it with incorrect input during verification, and after that, the loader keeps running and I can't retry the verification. That’s why I added this code here as well.

File: src/routes/signin/+page.svelte (hunk 4 above)
Comment (RaghvindYadav) [reply src/routes/signin/+page.svelte]: All these changes for mobile view.

File: src/routes/signup/sign-up-form.svelte (hunk 6 above)
Comment (RaghvindYadav) [reply src/routes/signup/sign-up-form.svelte]: @rupalirajput2597 Please answer the question here.

File: src/routes/signup/sign-up-form.svelte (hunk 7 above)
Comment (RaghvindYadav) [reply src/routes/signup/sign-up-form.svelte:442]: @rupalirajput2597 Please answer the question here too.

File: src/routes/reset-password/reset-password-form.svelte (hunk 3 above)
Comment (RaghvindYadav) [reply src/routes/reset-password/reset-password-form.svelte]: okay

File: src/routes/reset-password/reset-password-form.svelte (hunk 5 above)
Comment (RaghvindYadav) [reply src/routes/reset-password/reset-password-form.svelte]: Okay, I think I miss understood your [message](https://baragaun.slack.com/archives/C084NTKGT26/p1744671806246159) in the slack.

> Reset Password
We need to handle cases where the ident is not found. Right now it just fails, but the actual issue from the server is that the user can't be found

File: src/routes/signup/sign-up-form.svelte (hunk 6 above)
Comment (rupalirajput2597) [reply src/routes/signup/sign-up-form.svelte]: Hello @jonahtwalker , yes the this code is necessary, because the typeof result is not `string` it is `object`, for that we need to add this logic:

Screenshot 2025-04-17 at 11 25 20 AM

File: src/routes/signup/sign-up-form.svelte (hunk 7 above)
Comment (rupalirajput2597) [reply src/routes/signup/sign-up-form.svelte:442]: We need to send this to show/hide `new suggestion` button for ident and related stuff
